src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,550 @@
+	oval_definition_set_version(definition, 1);
+	oval_definition_set_title(definition, "CVRF RPM Vulnerability Test");
+	struct oval_criteria_node *criteria = oval_criteria_node_new(def_model, OVAL_NODETYPE_CRITERIA);
+	oval_definition_set_criteria(definition, criteria);
+
+	while (oscap_string_iterator_has_more(product_ids)) {
+		product_id = oscap_string_iterator_next(product_ids);
+		struct cvrf_rpm_attributes *rpm_attr = parse_rpm_name_into_components(eval, product_id);
+
+		struct oval_object *object = cvrf_definition_model_get_new_object(def_model, rpm_attr, index);
+		struct oval_state *state = cvrf_definition_model_get_new_state(def_model, rpm_attr, index);
+
+		struct oval_test *rpm_test = cvrf_definition_model_get_new_rpm_test(def_model, index);
+		oval_test_set_object(rpm_test, object);
+		oval_test_add_state(rpm_test, state);
+
+		struct oval_criteria_node *criterion = oval_criteria_node_new(def_model, OVAL_NODETYPE_CRITERION);
+		oval_criteria_node_set_test(criterion, rpm_test);
+		char *comment = oscap_sprintf("Check for vulnerability of package %s", rpm_attr->rpm_name);
+		oval_criteria_node_set_comment(criterion, comment);
+		oval_criteria_node_add_subnode(criteria, criterion);
+
+		cvrf_rpm_attributes_free(rpm_attr);
+		index++;
+	}
+	oscap_string_iterator_free(product_ids);
+
+	oval_definition_model_export(def_model, "cvrf-definition-model-export.xml");
+
+	/*
+	struct oval_syschar_model *syschar_model = oval_syschar_model_new(def_model);
+	struct oval_probe_session_t *probe_session = oval_probe_session_new(syschar_model);
+	oval_syschar_model_free(syschar_model);
+	oval_probe_session_free(probe_session);
+	*/
+	return 0;
+}
+
+struct oval_test *cvrf_definition_model_get_new_rpm_test(struct oval_definition_model *def_model, int testNo) {
+
+	struct oval_test *rpm_test = oval_test_new(def_model, get_oval_id_string("test", testNo));
+	oval_test_set_subtype(rpm_test,OVAL_LINUX_RPM_INFO);
+	oval_test_set_version(rpm_test, 1);
+	oval_test_set_check(rpm_test, OVAL_CHECK_AT_LEAST_ONE);
+	oval_test_set_existence(rpm_test, OVAL_AT_LEAST_ONE_EXISTS);
+
+	return rpm_test;
+
+}
+
+struct oval_object *cvrf_definition_model_get_new_object(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, int objectNo) {
+
+	const char *object_id;
+	object_id = get_oval_id_string("object", objectNo);
+
+	struct oval_object *object = oval_definition_model_get_new_object(def_model, object_id);
+	oval_object_set_subtype(object, OVAL_LINUX_RPM_INFO);
+	struct oval_object_content *object_content = oval_object_content_new(def_model, OVAL_OBJECTCONTENT_ENTITY);
+	struct oval_entity *object_entity = oval_entity_new(def_model);
+	oval_entity_set_name(object_entity, attributes->rpm_name);
+	oval_object_content_set_entity(object_content, object_entity);
+	oval_object_add_object_content(object, object_content);
+
+	return object;
+}
+
+struct oval_state *cvrf_definition_model_get_new_state(struct oval_definition_model *def_model,
```

Comment:
the method name implies this takes cvrf_definition_model but it takes oval_definition_model